File: Code Examples/Security_OTP_Example/src/main/java/io/hashimati/security/domains/RefreshToken.java
package io.hashimati.security.domains;

import io.micronaut.data.annotation.*;
import io.micronaut.data.model.naming.NamingStrategies;
import lombok.Data;

import java.time.Instant;

@Data
@MappedEntity(value = "refreshTokens", namingStrategy = NamingStrategies.Raw.class)
public class RefreshToken {
    @Id
    @GeneratedValue(GeneratedValue.Type.AUTO)
    private Long id;
    private String username;
    private String refreshToken;
    private Boolean revoked;
    @DateCreated
    private Instant dateCreated;

    @DateUpdated
    private Instant dateUpdated;
}

File: MicroCli/src/main/resources/micronaut/security/jwt/kotlin/mongoReactive/domains/RefreshToken.kt
package ${securityPackage}.domains

import java.time.Instant

data class RefreshToken (
    var id: String? = null,
    var username: String? = null,
    var refreshToken: String? = null,
    var revoked: Boolean? = null,
    var dateCreated: Instant? = null,
){
    
}